### PR TITLE
Add liveness/readiness probes

### PR DESCRIPTION
**Pull request by trisberg**
Resolves: #3908

### CODE REVIEW

File: src/kubernetes/server/server-deployment.yaml [hunk 1]
```diff
@@ -24,6 +24,16 @@ spec:
             readOnly: true
         ports:
         - containerPort: 80
+        livenessProbe:
+          httpGet:
+            path: /management/health
+            port: 80
+          initialDelaySeconds: 45
```
Comment (sabbyanandan) [inline src/kubernetes/server/server-deployment.yaml:31]: Just for consistency, I wonder if the time-values can be the same as what we have as defaults in the [K8s-deployer](https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/blob/0cb8e688633faea9377f5068600b183e14b6e703/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java#L377-L401).

Comment (jvalkeal) [reply src/kubernetes/server/server-deployment.yaml:31]: Btw, these changes need to go under `src/templates/kubernetes`. `src/README.md` has notes about this structure.

Comment (jvalkeal) [reply src/kubernetes/server/server-deployment.yaml:31]: ah, never mind as template was also updated.

Comment (trisberg) [reply src/kubernetes/server/server-deployment.yaml:31]: > Just for consistency, I wonder if the time-values can be the same as what we have as defaults in the [K8s-deployer](https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/blob/0cb8e688633faea9377f5068600b183e14b6e703/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java#L377-L401).

These apps take longer to start up than the stream apps, so we need to adjust for that

**Pull request merged**